### ISSUE TITLE
text-show-3.9.4 fails to compile with GHC-8.0.2

### ISSUE THREAD

**Issue by phadej** [RyanGlScott/text-show#50]
EDIT: I don't know if it's something text-show does, or text-2.0 (the install plan has text-2.0).

```
Resolving dependencies...
Build profile: -w ghc-8.0.2 -O1
In order, the following will be built (use -v for more details):
 - text-show-3.9.4 (lib) (first run)
Configuring library for text-show-3.9.4..
Preprocessing library for text-show-3.9.4..
Building library for text-show-3.9.4..
[ 1 of 66] Compiling TextShow.Utils   ( src/TextShow/Utils.hs, /codetmp/text-show-3.9.4/dist-newstyle/build/x86_64-linux/ghc-8.0.2/text-show-3.9.4/build/TextShow/Utils.o )
...
[33 of 66] Compiling TextShow.Data.Complex ( src/TextShow/Data/Complex.hs, /codetmp/text-show-3.9.4/dist-newstyle/build/x86_64-linux/ghc-8.0.2/text-show-3.9.4/build/TextShow/Data/Complex.o )
ghc: panic! (the 'impossible' happened)
  (GHC version 8.0.2 for x86_64-unknown-linux):
	Simplifier ticks exhausted
  When trying UnfoldingDone $
  To increase the limit, use -fsimpl-tick-factor=N (default 100)
  If you need to do this, let GHC HQ know, and what factor you needed
  To see detailed counts use -ddump-simpl-stats
  Total ticks: 26920

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```

**Comment by phadej** [comment]
With `{-# OPTIONS_GHC -fsimpl-tick-factor=200 #-}` text-show seems to compile fine with GHC-8.0.2

**Comment by RyanGlScott** [comment]
Good catch. I didn't notice this at the time of releasing `text-show-3.9.4`, since the CI job's build plan [picked `text-1.2.5.0`](https://github.com/RyanGlScott/text-show/runs/4638234494?check_suite_focus=true#step:15:49). Building `text-show-3.9.4` with `text-1.2.5.0` succeeds, but fails with `text-2.0`. Bummer.

I wonder what the most robust way to fix this issue is. It's very likely that `text-show` uses too many `INLINE` pragmas currently, which would exacerbate the issue. But without careful profiling, it's not obvious to me which `INLINE` pragmas to remove. I've previously worked around this issue by disabling optimizations in certain modules (see 0cba20205cfabead51d868ebc772a52971fb4027, for instance), but your `-fsimpl-tick-factor=200` suggestion is probably the better one. I should use that consistently.

**Comment by RyanGlScott** [comment]
I've opened #52 with a short-term fix, and I've opened #51 as a reminder to investigate the underlying issue more carefully.

**Comment by RyanGlScott** [comment]
I've uploaded [`text-show-3.9.5`](https://hackage.haskell.org/package/text-show-3.9.5) to Hackage with the aforementioned short-term fix.
